cast/src/lib.rs

```diff
@@ -1274,6 +1274,24 @@ impl SimpleCast {
         Ok(parse_bytes32_string(&buffer)?.to_owned())
     }
 
+    /// Decodes checksummed address from bytes32 value
+    pub fn parse_bytes32_address(s: &str) -> Result<String> {
+        let s = strip_0x(s);
+        if s.len() != 64 {
+            eyre::bail!("string not 32 bytes");
+        }
+
+        let s = strip_12_zero_bytes(s);
```

Comment:
no need for an extra function here,

we can inline the strip_prefix, use `if let Some()` and return an error in the else branch

Reply:
Changed in commit https://github.com/foundry-rs/foundry/pull/4746/commits/6ea39963b4456e808b8a7121f7e4ee614d647a98